tests/test_os.py

```diff
@@ -500,34 +503,94 @@ async def test_abspath():
     assert result == abs_filename
 
 
-@pytest.mark.parametrize(("path",), [(".",), ("..",)])
-async def test_walk(path: str):
-    """Test the `walk` call."""
+def _act_on_error(oserror):
+    print(f"aiofiles.os.walk onerror handler: {oserror}")
 
-    os_walk_res = list(walk(top=path))
-    aio_walk_res = [r async for r in aiofiles.os.walk(top=path)]
-
-    assert aio_walk_res == os_walk_res
 
+@pytest.mark.parametrize(
+    ("top", "topdown", "onerror", "followlinks"),
+    [
+        ("non-existent", True, _act_on_error, False),
+        ("README.md", True, None, False),
+        ("README.md", False, _act_on_error, False),
+        ("./src", True, _act_on_error, False),
+        ("./src", False, _act_on_error, False),
+        ("./tests", True, None, True),
+        ("./tests", False, None, True),
+    ],
+)
+async def test_walk(top, topdown, onerror, followlinks):
+    result = [
+        row
+        async for row in aiofiles.os.walk(
+            top=top, topdown=topdown, onerror=onerror, followlinks=followlinks
+        )
+    ]
+    answer = [
+        row
+        for row in os.walk(
+            top=top, topdown=topdown, onerror=onerror, followlinks=followlinks
+        )
+    ]
```

Comment:
Just comparing that the `async` version works in the same way as the `sync` one does.